src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1175 @@ const ActorInit En_Kakasi_InitVars = {
+        gSaveContext.weekEventReg[79] |= 8;
+        func_800B7298(globalCtx, &this->actor, 6);
+        ActorCutscene_Stop(this->actorCutscenes[0]);
+        this->aboveGroundStatus = ENKAKASI_ABOVE_GROUND_TYPE;
+        this->songSummonDist = 80.0f;
+        EnKakasi_SetupIdleUnderground(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970740.s")
+void EnKakasi_SetupIdleUnderground(EnKakasi* this) {
+    this->actor.shape.yOffset = -7000.0;
+    this->actor.draw = NULL;
+    this->actor.flags |= 0x8000000;
+    this->unkState196 = 5;
+    this->actionFunc = EnKakasi_IdleUnderground;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970978.s")
+void EnKakasi_IdleUnderground(EnKakasi* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.weekEventReg[79] & 0x8) && this->actor.xzDistToPlayer < this->songSummonDist &&
+        (gGameInfo->data[2401] != 0 || globalCtx->msgCtx.unk1202A == 0xD)) {
+        this->actor.flags &= ~0x8000000;
+        globalCtx->msgCtx.unk1202A = 4;
+        this->actionFunc = EnKakasi_SetupRiseOutOfGround;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970A10.s")
+void EnKakasi_SetupRiseOutOfGround(EnKakasi* this, GlobalContext* globalCtx) {
+    s32 cutsceneIndex;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970A9C.s")
+    cutsceneIndex = 0;
+    if (this->aboveGroundStatus == ENKAKASI_ABOVE_GROUND_TYPE) {
+        cutsceneIndex = 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970F20.s")
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(this->actorCutscenes[cutsceneIndex]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80970FF8.s")
+    } else if (ActorCutscene_GetCanPlayNext(this->actorCutscenes[cutsceneIndex]) == 0) {
+        ActorCutscene_SetIntentToPlay(this->actorCutscenes[cutsceneIndex]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971064.s")
+    } else {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actorCutscenes[cutsceneIndex], &this->actor);
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_AKINDONUTS_HIDE);
+        this->actor.draw = EnKakasi_Draw;
+        this->unkState196 = 6;
+        this->actionFunc = EnKakasi_RisingOutOfGround;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971430.s")
+void EnKakasi_RisingOutOfGround(EnKakasi* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x3000;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kakasi/func_80971440.s")
+    if (this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+    }
+    if (this->actor.shape.yOffset < -10.0f) {
+        if ((globalCtx->gameplayFrames & 7) == 0) {
```

Comment:
Would this match?
```suggestion
        if ((globalCtx->gameplayFrames % 8) == 0) {
```

Reply:
Fascinating that the math works like that.